### PR TITLE
NNFrames: support caching training data on Disk

### PR DESCRIPTION
**Pull request by hhbyyh**
Could be very helpful for training with enormous training data, before this we always cache data in mem.

### CODE REVIEW

**Comment by jason-dai** [comment]
Please update example/document

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/nnframes/NNEstimator.scala [hunk 1]
```diff
@@ -378,19 +400,20 @@ class NNEstimator[T: ClassTag] private[zoo] (
...
 
-    initialDataSet.transform(SampleToMiniBatch[T](batchSize))
+    initialDataSet.transform(SampleToMiniBatch[T](batchSize)).toDistributed()
   }
 
   protected override def internalFit(dataFrame: DataFrame): NNModel[T] = {
     val trainingDataSet = getDataSet(dataFrame, $(batchSize))
     val endTrigger = if (isSet(endWhen)) $(endWhen) else Trigger.maxEpoch($(maxEpoch))
-    val optimizer = Optimizer(model, trainingDataSet, criterion)
+    val optimizer = new InternalDistriOptimizer(model, trainingDataSet, criterion)
```
Comment (qiuxin2012) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/nnframes/NNEstimator.scala]: I think Estimator API is better.

Comment (hhbyyh) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/nnframes/NNEstimator.scala]: Estimator is not subclass of DistriOptimizer, and that would bring many changes.

Comment (qiuxin2012) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/nnframes/NNEstimator.scala]: @hhbyyh You should use the Estimator's train function at least. 
https://github.com/intel-analytics/analytics-zoo/blob/338fe3fe7760b262576eb4e4555e38a616722e3a/zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala#L1264

Or Disk_and_memory won't work.

**Comment by hhbyyh** [comment]
http://172.168.2.101:8080/view/ZOO-PR-PIPELINE/job/ZOO-PR-Validation/2627/ passed. Merging to master. Thanks for the review.

**Pull request merged**